### ISSUE TITLE
Trace output errors

### ISSUE THREAD

**Issue by slundberg** [JuliaOpt/Optim.jl#147]
It seems like show_trace is reporting the estimated gradient as the function value and not showing the function value at all. See the below example on master:

```julia
julia> optimize(x->norm(x+4)+4, [0.0,0.0], show_trace=true, iterations=20)
Iter     Function value   Gradient norm 
     0     3.517950e-01              NaN
     1     3.187842e-01              NaN
     2     6.838181e-01              NaN
     3     6.390792e-01              NaN
     4     1.360921e+00              NaN
     5     1.246015e+00              NaN
     6     2.464085e-01              NaN
     7     7.868933e-02              NaN
     8     3.687705e-01              NaN
     9     3.532858e-01              NaN
    10     1.366622e-01              NaN
    11     1.043260e-01              NaN
    12     4.241014e-02              NaN
    13     8.865064e-02              NaN
    14     8.766231e-02              NaN
    15     4.231915e-02              NaN
    16     1.743947e-02              NaN
    17     1.577730e-02              NaN
    18     1.009202e-02              NaN
    19     5.752459e-03              NaN
    20     3.368566e-03              NaN
Results of Optimization Algorithm
 * Algorithm: Nelder-Mead
 * Starting Point: [0.0,0.0]
 * Minimum: [-4.005367537339528,-3.998781104882558]
 * Value of Function at Minimum: 4.005504
 * Iterations: 20
 * Convergence: false
   * |x - x'| < NaN: false
   * |f(x) - f(x')| / |f(x)| < 1.0e-08: false
   * |g(x)| < NaN: false
   * Exceeded Maximum Number of Iterations: true
 * Objective Function Calls: 38
 * Gradient Call: 0
```

**Comment by pkofod** [comment]
I will add, that you cannot use `extended_trace=true` with nelder mead for this reason. We probably have to look at the internals of `NelderMead()` to fix syntax between this and other methods.

```julia
# This _won't_ work 
optimize(f, [0., 0.];method = :nelder_mead, store_trace=true, extended_trace=true)
```

**Comment by slundberg** [comment]
Okay. If there is some small change to the PR that would be helpful let me know. Otherwise we can just wait for a more thorough change.

**Comment by pkofod** [comment]
fixed by #204
